src/agent.spec.ts

```diff
@@ -0,0 +1,101 @@
+import { Finding, FindingSeverity, FindingType, HandleTransaction, ethers } from 'forta-agent';
+import { createAddress } from 'forta-agent-tools';
+import { TestTransactionEvent, MockEthersProvider } from 'forta-agent-tools/lib/test';
+import { provideHandleTransaction } from './agent';
+import { swapEvent, uniswapPoolABI, UNI_INIT_CODE, UNISWAP_FACTORY_ADDRESS } from './agent'; // Import ABI and constants
+import { computePoolAddress } from './agent'; // Import computeAddress function
+
+describe('Uniswap V3 Swap Event Detector', () => {
```

Comment:
I'd suggest adding one more test for completion "it returns multiple Findings when there are multiple Uniswap V3 swaps"